tests/contrib/grpc/test_grpc.py

```diff
@@ -30,6 +33,25 @@ def tearDown(self):
         unpatch()
         super(GrpcTestCase, self).tearDown()
 
+    def get_spans(self, size=0, retry=20):
+        # testing instrumentation with grpcio < 1.14.0 presents a problem for
```

Comment:
I don't have enough information here to know what is going on.

What is the tl;dr; of the issue and why do we need a retry?

Reply:
In grpc <= 1.13, grpc returns the call before the future callbacks are finished.
In the tests, we check if the spans are in your fake writer just after the call, but there might not be as the futures callback are not finished.

The best strategy to sync the thread is to do a sleep-and-wait.